.github/github_workflow_scripts/purge_branch_protection_rules.py

```diff
@@ -0,0 +1,358 @@
+    """
+    Helper method to convert the response to a list
+    of `BranchProtectionRule`.
+
+    Arguments:
+    - `response` (``dict[str, Any]``): The response data.
+
+    Returns:
+    - a `list[BranchProtectionRule]`.
+
+    Raises:
+    - `KeyError | AttributeError` in case the conversion fails.
+    """
+
+    rules: list[BranchProtectionRule] = []
+
+    try:
+        for node in response.get('data', {}).get('repository').get('branchProtectionRules').get('nodes'):
+            rule = BranchProtectionRule(
+                id=node.get("id"),
+                pattern=node.get("pattern"),  # ignore: type
+                matching_refs=node.get("matchingRefs").get("totalCount")
+            )
+
+            rules.append(rule)
+    except (KeyError, AttributeError) as e:
+        raise e.__class__(f"{e.__class__.__name__} parsing '{response=}' as a branch protection rule: {e}")
+
+    return rules
+
+
+def should_delete_rule(rule: BranchProtectionRule) -> bool:
+    """
+    Check whether we should delete this rule.
+    To determine if we should delete the rule we check that:
+
+    * The rule is not in the list of protected rules (see `self.PROTECTED_RULES`)
+    * The rule does not apply to any branches.
+
+    Returns:
+    - `True` if we can delete the rule, `False` otherwise.
+    """
+
+    should = False
+
+    if rule.pattern in PROTECTED_RULES:
+        logger.info(f"{rule} not deleted because it's in the list of protected rules '{','.join(PROTECTED_RULES)}'")
+    elif rule.matching_refs > 0:
+        logger.info(f"{rule} not deleted because it's associated to {rule.matching_refs} existing branches/refs")
+    else:
+        should = True
+
+    return should
+
+
+def write_deleted_summary_to_file(deleted: list[BranchProtectionRule]) -> None:
+    """
+    Helper function to create a Markdown summary file for deleted branches
+    if the `GITHUB_STEP_SUMMARY` is set. See
+    https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary
+
+    Arguments:
+    - `deleted` (``list[BranchProtectionRule]``): A list of deleted `BranchProtectionRule`s.
+    """
+
+    if os.getenv(GH_JOB_SUMMARY_ENV_VAR):
+        fp = Path(str(os.getenv(GH_JOB_SUMMARY_ENV_VAR)))
+
+        if not fp.is_file():
+            logger.warning(f"The env var {GH_JOB_SUMMARY_ENV_VAR} is not set to a file. Not writing summary...")
+            return
+
+        header = "## Deleted Branch Protection Rules"
+
+        if not deleted:
+            body = "### No branch protection rules were deleted"
+            markdown_content = f"{header}\n\n{body}\n"
+        else:
+            table_header = "| ID | Pattern | Matching Refs |\n| --- | ------- | ------------- |"
+            table_rows = [f"| {rule.id} | {rule.pattern} | {rule.matching_refs} |" for rule in deleted]
+            table_body = "\n".join(table_rows)
+            markdown_content = f"{header}\n\n{table_header}\n{table_body}\n"
```

Comment:
please use `tabulate` https://pypi.org/project/tabulate/ it's in pyproject.toml CI section.
![image](https://github.com/user-attachments/assets/1e3ecac9-3ecc-4096-825c-2f95ef312196)

Reply:
https://github.com/demisto/content/pull/35604/commits/5d88ebc49282be5616255597dda6a53a2a1dd944